tools/bin/mbedtls-rewrite-branch-style

```diff
@@ -0,0 +1,551 @@
+# limitations under the License.
+
+import argparse
+import os
+import pathlib
+import subprocess
+import sys
+from typing import List, Optional
+
+
+class BranchRewriter:
+    """Tools to rebase and rewrite a Git branch."""
+
+    GIT_EXE = 'git'
+
+    def __init__(self,
+                 working_branch_name: Optional[str] = None,
+                 verbose=False) -> None:
+        self.verbose = verbose
+        self.original_worktree_path = None # type: Optional[str]
+        self.worktree_path = None # type: Optional[str]
+        self.working_branch_name = \
+            working_branch_name if working_branch_name else None
+        self.rewritten_sha = None #type: Optional[str]
+        self.ok = True
+
+    def info(self, fmt, *args, **kwargs) -> None:
+        """In verbose mode, print a formatted message."""
+        if self.verbose:
+            sys.stderr.write(fmt.format(*args, **kwargs) + '\n')
+
+    def log_error(self, fmt, *args, **kwargs) -> None:
+        """Print a formatted message and remember that an error occurred."""
+        self.ok = False
+        sys.stderr.write(fmt.format(*args, **kwargs) + '\n')
+
+    def run_git(self, git_cmd: List[str], **kwargs) -> str:
+        """Run a git command and return its output."""
+        if 'universal_newlines' not in kwargs:
+            kwargs['universal_newlines'] = True
+        cmd = [self.GIT_EXE] + git_cmd
+        return subprocess.check_output(cmd, **kwargs)
+
+    def check_git(self, git_cmd: List[str], **kwargs) -> bool:
+        """Run a git command for its boolean result.
+
+        Return True if the command returns 0 and False if it returns 1.
+        Raise subprocess.CalledProcessError for any other status.
+        """
+        cmd = [self.GIT_EXE] + git_cmd
+        cp = subprocess.run(cmd, check=False, **kwargs)
+        if cp.returncode == 0:
+            return True
+        elif cp.returncode == 1:
+            return False
+        else:
+            raise subprocess.CalledProcessError(cp.returncode, cmd)
+
+    def run_git_checkout(self, revision: str) -> None:
+        """Check out the given revision in detached mode."""
+        self.run_git(['checkout', '--detach', '--quiet', revision])
+
+    def get_git_lines(self, git_cmd: List[str], **kwargs) -> List[str]:
+        """Run a git command and return its output, split into lines.
+
+        Ignore trailing empty lines, but preserve leading and inner empty lines.
+        """
+        output = self.run_git(git_cmd, **kwargs)
+        return output.rstrip('\n').split('\n')
+
+    def head_sha(self) -> str:
+        """The sha of the git HEAD."""
+        return self.run_git(['rev-parse', 'HEAD']).strip()
+
+    def checked_out_branch(self) -> Optional[str]:
+        """The name of the branch checked out in the Git working directory.
+
+        None if no branch is checked out (detached HEAD).
+        """
+        branch_name = self.run_git(['rev-parse', '--abbrev-ref', 'HEAD']).strip()
```

Comment:
Out of curiosity (for my own education): I usually use `git branch --show-current` for this, is using `rev-parse` more robust?

Reply:
I didn't know `git branch --show-current`. It's more recent: `rev-parse --abbrev-ref` has been around for a very long time, but `git branch --show-current` was introduced between Ubuntu 18.04 and 20.04.

Reply:
Ok, thanks!